src/components/SearchResultsList/SearchResultsList.tsx

```diff
@@ -0,0 +1,101 @@
+import { Outlet, useNavigate } from 'react-router-dom';
+import { Animal, AnimalsPagedQueryResponse } from '../../services/api';
+import styles from './SearchResultsList.module.css';
+
+interface SerachResultsListProps {
+  animalsResponseData: AnimalsPagedQueryResponse;
+  setPage: (pageNumber: string | null) => void;
+}
+
+export function SearchResultsList(props: SerachResultsListProps) {
+  const navigate = useNavigate();
+
+  const getAnimalType = (animal: Animal) => {
+    const keysToSkip: Array<keyof Animal> = ['uid', 'name'];
+    return (Object.keys(animal) as Array<keyof Animal>)
+      .filter((typeKey) => animal[typeKey] && !keysToSkip.includes(typeKey))
+      .map((typeKey) =>
+        typeKey.replace(/([a-z])([A-Z])/, `$1 $2`).toLowerCase(),
+      )
+      .join(', ');
+  };
+
+  const clickListHandler = (e: React.MouseEvent) => {
+    if (e.target instanceof HTMLElement) {
+      if (location.pathname.includes('details')) {
+        navigate({
+          pathname: '/',
+          search: location.search,
+        });
+        return;
+      }
+      const target = e.target.closest('li');
+      if (!target) {
+        return;
+      }
+      const uid = target.getAttribute('id');
+      if (target && uid) {
+        navigate({
+          pathname: `details/${uid}`,
+          search: location.search,
+        });
+      }
+    }
+  };
+
+  const renderListHeader = () => {
+    return (
+      <li className={`${styles.searchResultsListItem} ${styles.header}`}>
+        <div className={`${styles.searchResultsListItemCell} ${styles.colOne}`}>
+          Name
+        </div>
+        <div className={`${styles.searchResultsListItemCell} ${styles.colTwo}`}>
+          Uid
+        </div>
+        <div
+          className={`${styles.searchResultsListItemCell} ${styles.colThree}`}
+        >
+          Type
+        </div>
+      </li>
+    );
+  };
+
```

Comment:
as for me, there are too many different logic in this component, which is not directly connected to it
my suggestion is to move all functions that are not using any props in a separate file like `SearchResultsList.helpers.tsx` (`getAnimalType`), and move this `renderSomething` functions into a separate components (`<ListHeader />` and `<ListItems />`) and pass required data as props to them

Reply:
just FYI - there is an approach, when all complex logic and render are separated. to do this you can move all isolated functions (which are not using props from lexical env, and read data only from props) to .helpers, and move all functions that actually depend on props into separate hook for this specific screen.
We can discuss it a bit more on the call, if I'll not forgot)

Reply:
Fixed.